### PR TITLE
[Matrix] remove p8-platform, major cleanups, settings.xml rework, LICENSE add

### PR DESCRIPTION
**Pull request by AlwinEsch**
Has done a major update of the addon.

But was only possible to do a load / start test (where OK), but have no access to related source!

@phunkyfish, @ksooo or someone else can you do a check about his work?

Thought to kick out the p8-platform and shot out this tabs use in source (done by Kodi `clang` call)

Also are the same things like on others about LICENSE (issue related) and assets (Kodi request related) added.

### CODE REVIEW

File: src/PVRFilmonData.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ PVRFilmonData::~PVRFilmonData(void) {
 }
 
 bool PVRFilmonData::Load(std::string user, std::string pwd) {
-	P8PLATFORM::CLockObject lock(m_mutex);
+	std::unique_lock<std::recursive_mutex> lock(m_mutex);
```
Comment (phunkyfish) [inline src/PVRFilmonData.cpp]: Any reason for using this or would the simpler `std::lock_guard<std::mutex>` have worked instead?

I'm just curious!

Comment (AlwinEsch) [reply src/PVRFilmonData.cpp]: Not direct, only was the `std::recursive_mutex` more equal to the way used by p8-platform.

Comment (AlwinEsch) [reply src/PVRFilmonData.cpp]: Is updated, the recursive_mutex makes no sense by this calls.

**Comment by phunkyfish** [comment]
We could ask on the forum to see if any users could test the PR?

The changes look correct to me. It's similar to what I did in vuplus, but probably safer to have a real runtime test.

**Comment by phunkyfish** [comment]
I posted on the forum let's see if someone responds.

EDIT: Got this tested, it works.

**Review by phunkyfish — APPROVED** [review]
LGTM

**Pull request merged**